pkg/controllers/user/approuter/ingress_controller.go

```diff
@@ -44,6 +43,10 @@ func isGeneratedDomain(obj *extensionsv1beta1.Ingress, host, domain string) bool
 }
 
 func (c *Controller) sync(key string, obj *extensionsv1beta1.Ingress) error {
+	if obj == nil {
```

Comment:
should we check for obj.DeletionTimestamp != nil too? both can indicate the removed object

Reply:
also I'd think that underlying code should cleanup dns record for the ingress that had been removed, right? even if the sync method returns

Reply:
The rdns-server(coredns) can support wildcard records, so only one record will be stored for a k8s cluster,all the ingress resources in a cluster use the dns record.The record will be automatically removed only after the k8s cluster is destroyed.

Reply:
@alena1108 yes,It's really necessary to check `obj.DeletionTimestamp != nil` and I have added it.

Reply:
Nit: can we do it in one line? if `obj == nil ||`. Also `obj.DeletionTimestamp` - you can access directly, no need to reference via objectMeta.